lib/commanded/event/error_handler.ex

```diff
@@ -0,0 +1,13 @@
+defmodule Commanded.Event.ErrorHandler do
+  alias Commanded.Event.FailureContext
+
+  @type failure_context :: map() | %FailureContext{context: map}
+  @type reason :: term()
+  @type delay :: non_neg_integer()
+
+  @callback handle_error({:error, reason}, struct(), %FailureContext{context: map}) ::
+              :skip
+              | {:retry, failure_context}
+              | {:retry, delay, failure_context}
+              | {:stop, reason}
```

Comment:
```suggestion
  @callback handle_error({:error, reason}, struct(), %FailureContext{context: map}, opts: keyword()) ::
              :skip
              | {:retry, failure_context}
              | {:retry, delay, failure_context}
              | {:stop, reason}
```

you will probably require some configuration for the strategy

Reply:
I want to keep it simple, if you have advanced needs: specify a module or override on_error/3